common/client-core/src/init/helpers.rs

```diff
@@ -132,25 +132,25 @@ impl<'a, G: ConnectableGateway> GatewayWithLatency<'a, G> {
     }
 }
 
-pub async fn gateways_for_init<R: Rng>(
-    rng: &mut R,
+pub async fn gateways_for_init(
     nym_apis: &[Url],
     user_agent: Option<UserAgent>,
     minimum_performance: u8,
     ignore_epoch_roles: bool,
 ) -> Result<Vec<RoutingNode>, ClientCoreError> {
-    let nym_api = nym_apis
-        .choose(rng)
-        .ok_or(ClientCoreError::ListOfNymApisIsEmpty)?;
+    // Build client with ALL URLs for fallback support
+    let nym_api_urls: Vec<nym_http_api_client::Url> = nym_apis
+        .iter()
+        .filter_map(|url| nym_http_api_client::Url::parse(url.as_str()).ok())
```

Comment:
```suggestion
        .map(nym_http_api_client::Url::from)
```

there's an infallible conversion from `url::Url` into `nym_http_api_client::Url`